backend/de.metas.salescandidate.base/src/main/java/de/metas/ordercandidate/api/OLCandsProcessorExecutor.java

```diff
@@ -341,4 +343,43 @@ private final boolean isEligibleOrLog(final OLCand cand)
 
 		return true;
 	}
+
+	private void validateCompensationGroupKey(final ListMultimap<ArrayKey, OLCand> grouping)
+	{
+		final List<ArrayKey> invalidGroupingKeys = grouping.keySet()
+				.stream()
+				.filter(key -> !isValidGroupCompensation(grouping.get(key)))
+				.collect(Collectors.toList());
+		grouping.keySet()
+				.stream()
+				.filter(invalidGroupingKeys::contains)
+				.map(grouping::get)
+				.flatMap(Collection::stream)
+				.forEach(this::invalidateCandidate);
+	}
+
+	private void invalidateCandidate(final OLCand olCand)
+	{
+		olcandBL.markAsError(userInChargeId, olCand, new AdempiereException("Multiple candidates in this group have the same "));
```

Comment:
please add a user-understandable EN and DE error message

Reply:
Changed and added.